pkg/minikube/cluster/cluster.go

```diff
@@ -140,11 +144,23 @@ func DeleteHost(api libmachine.API) error {
 
 // Un-register host VM
 func unregister(host *host.Host) error {
+	if !minishiftConfig.InstanceConfig.Register {
+		return nil
+	}
 	if drivers.MachineInState(host.Driver, state.Running)() {
 		if err := registration.UnregisterHostVM(host, RegistrationParameters); err != nil {
 			return fmt.Errorf("Error unregistring the VM: %s", err)
 		}
 	}
+	if minishiftConfig.InstanceConfig.Register {
+		userConfirmation := util.ReadInputFromStdin("Current Minishift instance is registered with RHN, " +
+			"Still want to Delete/Stop VM (y/n)")
+		if strings.ToUpper(userConfirmation) == "Y" {
```

Comment:
This is not right. As we are asking whether user wants to unregister after UnregisterHostVM(host, RegistrationParameters)